coral/src/app/features/topics/details/documentation/TopicDocumentation.tsx

```diff
@@ -1,26 +1,94 @@
-import { PageHeader } from "@aivenio/aquarium";
+import { Alert, Box, PageHeader, useToast } from "@aivenio/aquarium";
 import { NoDocumentationBanner } from "src/app/features/topics/details/documentation/components/NoDocumentationBanner";
 import { useTopicDetails } from "src/app/features/topics/details/TopicDetails";
-import { DocumentationView } from "src/app/features/topics/details/documentation/components/DocumentationView";
+import { useState } from "react";
+import { useMutation, useQueryClient } from "@tanstack/react-query";
+import { updateTopicDocumentation } from "src/domain/topic";
+import { parseErrorMsg } from "src/services/mutation-utils";
+import { DocumentationEditor } from "src/app/components/documentation/DocumentationEditor";
+import { DocumentationView } from "src/app/components/documentation/DocumentationView";
 
 function TopicDocumentation() {
+  const queryClient = useQueryClient();
+
+  const [editMode, setEditMode] = useState(false);
+  const [saving, setSaving] = useState(false);
+
   const { topicOverview } = useTopicDetails();
 
+  const toast = useToast();
+
+  const { mutate, isError, error } = useMutation(
+    (markdownString: string) => {
+      setSaving(true);
+      return updateTopicDocumentation({
+        topicName: topicOverview.topicInfo.topicName,
+        topicIdForDocumentation: topicOverview.topicIdForDocumentation,
+        topicDocumentation: markdownString,
+      });
+    },
+    {
+      onSuccess: () => {
```

Comment:
Maybe we could try to stay ahead of the curve and stop using the query callbacks if they will be eventually deprecated? :o Like, have a `useEffect` running this when `status === 'success'` or something :o

Reply:
Sounds like a good refactor and maybe even a good first issue ticket?? 
I would not do it here, since the PR is already pretty big 🙈

Reply:
I mean, it's not that big if a change, is it? It's moving this block in a `useEffect` instead of `onSuccess` ^^ But you are right that it is a good first issue, so let's leave it like this.